src/odemis/driver/hamamatsurx.py

```diff
@@ -198,7 +198,7 @@ def _updateWavelengthList(self, _=None):
         """
         npixels = self._resolution[0]  # number of pixels, horizontal is wavelength
         # pixelsize VA is sensor px size without binning and magnification
-        pxs = self.pixelSize.value[0] * self._binning[0] * self._metadata.get(model.MD_LENS_MAG, 1.0)
+        pxs = self.pixelSize.value[0] * self._binning[0] / float(self._metadata.get(model.MD_LENS_MAG, 1.0))
```

Comment:
Why do you need to use `float()` here? If it's because of some trick, you'd need to put an extra comment. Otherwise, if it's just out of habbit, you can drop it: the division from the "future" takes care of always return floats when needed.

Reply:
I was worried that if in the yaml file an int is defined, it will do an integer division and that leads to incorrect pixelsize calculations. Or will every value in the yaml file automatically be read as a float?

Reply:
In Python 2, `5/7 == 0` (ie, an int divided by an int always return an int), *unless* you import the division from the "future". "future" actually means "Python 3". In Python 3, `5/7 == 5.0/7.0` (ie, an int divided by an int will return a float).
As we use the division from the "future", you never need to worry about it, and you can keep the code readable :-)

PS: in case you need to get an int out of a division, you can use `//`. For example: `5//7 == 0`.

Reply:
Thanks!